sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/resolver/NameScope.scala

```diff
@@ -403,6 +414,22 @@ class NameScope(
    *
    * {{ SELECT col1 + col2 AS a FROM VALUES (1, 2) GROUP BY a; }}}
    *
+   * In case we are resolving names in expression trees from HAVING or ORDER BY on top of
+   * [[Aggregate]], we are able to resolve hidden attributes only if theose are present in
```

Comment:
either 
```suggestion
   * In case we are resolving names in expression trees from [[Having]] or [[Sort]] on top of
   * [[Aggregate]], we are able to resolve hidden attributes only if theose are present in
```
or
```suggestion
   * In case we are resolving names in expression trees from HAVING or ORDER BY on top of
   * GROUP BY, we are able to resolve hidden attributes only if theose are present in
```
sound better (for consistency).

Reply:
1. There's no `Having` node.
2. `Aggregate` is not necessarily from GROUP BY, because we have global aggs.